bokehjs/src/coffee/models/sources/geojson_data_source.coffee

```diff
@@ -18,7 +18,7 @@ class GeoJSONDataSource extends ColumnDataSource.Model
   initialize: (options) ->
     super(options)
     @geojson_to_column_data() # this just validates the initial geojson value
-    @register_property('data', @geojson_to_column_data, true)
+    @define_computed_property('data', @geojson_to_column_data, true)
```

Comment:
@mattpap 

> @bokeh/dev, can someone explain why GeoJSONDataSource inherits from ColumnDataSource, yet overrides its data property with a computed data property? @bryevdv, your comment above indicates you were confused about this as well.

It's not ideal. We had a bigger plan to support GeoJSON (and other non-columnar data sources) properly but everything in BokehJS is currently very structured around the CDS so this compromise of sending the GeoJSON down the pipe and then turning it into columnar data on the BokehJS side is what we came up with.

I've been trying to dig up the long discussion I had with bryan about it, but can't find it at the moment.

But there's this which describes some of the plans: https://github.com/bokeh/bokeh/wiki/Geo-Ramblings#renderers--non-columnar-data-sources